src/store/VideoReducer.js

```diff
@@ -0,0 +1,39 @@
+const VideoReducer = (state, action) => {
+  switch (action.type) {
+    case 'SET_CURRENT_VIDEO':
+      return { ...state, currentVideo: action.payload };
+    case 'ADD_FAVORITE_VIDEO': {
+      const videoExists = () => {
+        return state.favoriteVideos.find((video) => {
+          return action.payload.videoId === video.videoId;
+        });
+      };
+      const videosIsNotEmpty = () => state.favoriteVideos.length !== 0;
+      if (videosIsNotEmpty() && videoExists()) {
+        return {
+          ...state,
+          favoriteVideos: state.favoriteVideos,
+        };
+      }
+      localStorage.setItem(
```

Comment:
I think that is better if inside the component that is updating the state you update the local storage, It was a convention in ```Redux``` to use only pure functions inside the reducers and I think it a good convention to follow with ```React``` as well.